test/discrepancies/Nonce.js

```diff
@@ -299,12 +342,586 @@ describe('@discrepancies - Nonce Test Suite', async () => {
+    const contractFactoryMnAfter = await getMirrorNodeNonce(internalCalleeContract.target);
+
+    //After successful deploy verify that the nonces of the factory contract and the tx signer are updated
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+    expectIncrementedNonce(0, 0, contractFactorySnAfter, contractFactoryMnAfter);
+
+    const selfdestructTx = await internalCalleeContract.selfdestructSample(deployedTempContractAddress);
+    await selfdestructTx.wait();
+
+    //signer nonce should increment after selfdestruct is executed
+    expectIncrementedNonce(snAfter, mnAfter, snAfter + 1, mnAfter + 1);
+
+    //redeploy the contract with the same salt and verify nonces are updated
+    const redeployTx = await internalCalleeContract.deployViaCreate2(1);
+    const redeployTxReceipt = await redeployTx.wait();
+
+    const redeployedTempContractAddress = redeployTxReceipt.logs.filter(
+      (e) => e.fragment.name === Constants.Events.DeployedContractAddress
+    )[0].args[0];
+
+    //get all addresses and verify nonces are updated
+
+  });
+
+  //NONCE-021
+  it('should not update nonces when deploying on an address with an already existing account', async function () {
+    //create a hollow account and send tokens
+    const wallet = await createNewAccountWithBalance(ethers.parseEther('3.1'));
+    const snAfterCreate = await getServicesNonce(wallet.address);
+    const mnAfterCreate = await getMirrorNodeNonce(wallet.address);
+    // verify nonce of hollow account after token received is 0
+    expectNonIncrementedNonce(snAfterCreate, mnAfterCreate, 0, 0);
+
+    //send ethers from the hollow account to the GENESIS account 
+    const signerFundTx = await wallet.sendTransaction({
+      to: signers[0].address,
+      value: Utils.tinybarToWeibarCoef // 1 tinybar
+    });
+    await signerFundTx.wait();
+
+    //verify new account nonce is 1
+    const snAfterSendTx = await getServicesNonce(wallet.address);
+    const mnAfterSendTx = await getMirrorNodeNonce(wallet.address);
+
+    expectIncrementedNonce(snAfterCreate, mnAfterCreate, snAfterSendTx, mnAfterSendTx);
+
+    //delete the account
+    const info = await Utils.getAccountInfo(wallet.address, sdkClient);
+    const deleteTx = await Utils.deleteAccount(wallet, sdkClient, info.accountId);
+
+    //send tokens to the same address and verify nonce is 0
+    const fundTx2 = await signers[0].sendTransaction({
+      to: wallet.address,
+      value: Utils.tinybarToWeibarCoef // 1 tinybar
+    });
+    await fundTx2.wait();
+
+    //verify nonce is 0
+    const snAfterNewCreate = await getServicesNonce(wallet.address);
+    const mnAfterNewCreate = await getMirrorNodeNonce(wallet.address);
+
+    expectNonIncrementedNonce(snAfterNewCreate, mnAfterNewCreate, 0, 0);
+  });
+
+  //NONCE-022
+  //should fix the get contract nonce function - waiting for SDKs
+  xit('should update all nonces when chained deploys of contracts', async function () {
```

Comment:
Q: why is this disabled?

Reply:
There are few tests where we are not able to get nonce of a smart contract from the sdk - this issue: https://github.com/hashgraph/hedera-sdk-js/issues/2192 . Basically we cannot assert that the nonce of a smart contract is correct